lib/python/qmk/info.py

```diff
@@ -833,24 +834,25 @@ def unknown_processor_rules(info_data, rules):
     return info_data
 
 
-def merge_info_jsons(keyboard, info_data):
+def merge_info_jsons(keyboard, info_data, skip_validation=False):
     """Return a merged copy of all the info.json files for a keyboard.
     """
     for info_file in find_info_json(keyboard):
         # Load and validate the JSON data
-        new_info_data = json_load(info_file)
+        new_info_data = json_load_cached(info_file)
```

Comment:
Here the existing cache implementation cannot be used, because the subsequent code modifies `new_info_data`, and those modifications would affect the data stored in the cache.
```suggestion
        new_info_data = json_load(info_file)
```

Reply:
Another option is to use the cache and make a `deepcopy` here, but we also would need to evaluate whether it would actually give some performance gain here.

(Adding that `deepcopy` to the caching wrapper itself would prevent any kind of errors due to modification of cached objects, but may consume lots of memory, so it's probably not the best solution.)

Reply:
`deepcopy` may well nullify speed improvements anyway.